kernel/register.cc

```diff
@@ -979,18 +979,6 @@ struct EchoPass : public Pass {
 	}
 } EchoPass;
 
-SatSolver *yosys_satsolver_list;
-SatSolver *yosys_satsolver;
-
-struct MinisatSatSolver : public SatSolver {
```

Comment:
I am not sure we want this removed. This allowed for a plugin to register a different SAT solver, though I am not sure `yosys_satsolver_list` gets used anywhere, so that the other-than-default solver would ever get selected.

Reply:
Looks like we are missing a command `satselect`, `solverselect` or similar which would update the `yosys_satsolver` pointer, but the overall approach of these `SatSolver` descendants appears sane. On the contrary, instead of removing this, we should arrange for an `ezcommand` solver to be instantiable through this.

Reply:
Thanks for your explanation, now I see the benefit. I thought it was entirely unused.
I can just remove this commit and the next one: ezSATCommand would not be accessible from the global solver yet, but would at least be available from C++ plugins.

Reply:
OK

Reply:
That's a good point. I will defer to @jix's judgement on this. Should we e.g. always override the default solver with the one from a plugin, if available?

Reply:
It would be nice to have a way to override the default solver, that would also make it easier to test this PR. I don't want to change the default without an explicit command though (`sat -select-solver ...` could be added, there is precedent for subcommand like options and it would avoid cluttering the command list, but maybe also make it a bit hard to discover)